subsys/bluetooth/host/iso.c

```diff
@@ -750,6 +724,77 @@ static uint16_t iso_chan_max_data_len(const struct bt_iso_chan *chan)
 	return max_data_len;
 }
 
+bool bt_iso_has_data(struct bt_conn *conn)
+{
+	return !k_fifo_is_empty(&conn->iso.txq);
+}
+
+struct net_buf *iso_data_pull(struct bt_conn *conn, size_t amount)
+{
+	LOG_DBG("iso-pull");
+
+	/* Leave the PDU buffer in the queue until we have sent all its
+	 * fragments.
+	 */
+	struct net_buf *frag = k_fifo_peek_head(&conn->iso.txq);
+
+	if (!frag) {
+		LOG_DBG("signaled ready but no frag available");
+		return NULL;
+	}
+
+	if (conn->iso.chan->state != BT_ISO_STATE_CONNECTED) {
+		LOG_DBG("channel has been disconnected");
+		struct net_buf *b = k_fifo_get(&conn->iso.txq, K_NO_WAIT);
+		(void)b;
+		__ASSERT_NO_MSG(b == frag);
+		return NULL;
+	}
+
+	if (bt_buf_has_view(frag)) {
+		/* This should not happen. conn.c should wait until the view is
+		 * destroyed before requesting more data.
+		 */
+		LOG_DBG("already have view");
+		return NULL;
+	}
+
+	bool last_frag = amount >= frag->len;
+
+	if (last_frag) {
+		LOG_DBG("last frag, pop buf");
+		struct net_buf *b = k_fifo_get(&conn->iso.txq, K_NO_WAIT);
+		(void)b;
+		__ASSERT_NO_MSG(b == frag);
+	}
+
+	return frag;
+}
+
+int conn_iso_send(struct bt_conn *conn, struct net_buf *buf, bool has_ts)
+{
+	if (buf->user_data_size < CONFIG_BT_CONN_TX_USER_DATA_SIZE) {
+		LOG_ERR("not enough room in user_data %d < %d pool %u",
+			buf->user_data_size,
+			CONFIG_BT_CONN_TX_USER_DATA_SIZE,
+			buf->pool_id);
+		return -EINVAL;
+	}
+
+	/* push the TS flag on the buffer itself.
+	 * It will be popped and read back by conn before adding the ISO HCI header.
+	 */
+	net_buf_push_u8(buf, has_ts ? 1 : 0);
```

Comment:
I guess this answers my previous comment about `1`. We should use a `define` for flags such as this

Reply:
added an enum